### ISSUE TITLE
mtbl doesn't build on OS X

### ISSUE THREAD

**Issue by mschiffm** [farsightsec/mtbl#2]
OS X lacks POSIX real-time clock _POSIX_TIMERS stuff.  Therefore, this happens:

```
$ make
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
  CC       mtbl/fileset.lo
mtbl/fileset.c:189:8: warning: implicit declaration of function 'clock_gettime'
      is invalid in C99 [-Wimplicit-function-declaration]
        res = clock_gettime(CLOCK_MONOTONIC, &now);
              ^
mtbl/fileset.c:189:22: error: use of undeclared identifier 'CLOCK_MONOTONIC'
        res = clock_gettime(CLOCK_MONOTONIC, &now);
                            ^
1 warning and 1 error generated.
make[1]: *** [mtbl/fileset.lo] Error 1
make: *** [all] Error 2
```

Two solutions:
- The function in question `mtbl_fileset_reload()`, only requires Unix Epoch-level timer resolution. We could substitute for `gettimeofday()` and wrap the whole stanza in conditional `#if` statements and achieve the same functionality.
- Fold in the ported real-time clock stuff as per `libnmsg`.

This begs another question: if `libmtbl` only needs second-level timer granularity, why use the POSIX-only real-time clock stuff in the first place?

This appears to be the only gating issue in getting mtbl ported to OS X.

**Comment by brudis-r7** [comment]
Any chance this is going to get OS X support?

**Comment by brudis-r7** [comment]
Fixed in #21
